wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WLabel.java

```diff
@@ -110,7 +111,11 @@ public String getText() {
 	 * @param args optional arguments for the message format string.
 	 */
 	public void setText(final String text, final Serializable... args) {
-		getOrCreateComponentModel().text = I18nUtilities.asMessage(text, args);
+		String currText = getText();
+
```

Comment:
This would probably work 90% of the time, but the value that should be tested against the current value is the response from 'asMessage(text,args)`.

Reply:
Just to clarify - you mean something like this?
```
String currText = getText();
Serializable textToBeSet = I18nUtilites.asMessage(text, args);
if (!Objects.equals(textToBeSet, currText) {
    getOrCreateComponentModel().text = textToBeSet;
}
```

Reply:
This is actually a bit tricky (or maybe a better term is messy).

The utility method `asMessage` can return either a `null`, `String` or `Message`.

`getText` always returns a `String`. If the `model` contains a `Message`, then `getText` transforms it into a `String`.

So to actually check if the `value` is being changed on the `model`, you cannot use `getText` but you have to get the value directly from the model:-

```
Serializable currText = getComponentModel().text;
Serializable textToBeSet = I18nUtilites.asMessage(text, args);
if (!Objects.equals(textToBeSet, currText)) {
    getOrCreateComponentModel().text = textToBeSet;
}
```